### ISSUE TITLE
HyperTreeList: resizing columns has black vertical bar in the wrong place

### ISSUE THREAD

**Issue by JillHolliday** [wxWidgets/Phoenix#1901]
**Operating system**: Windows 10
**wxPython version & source**:          4.1.1, 4.1.0 PyPI,   3.0.2.0 downloaded from SourceForge IIRC
**Python version & source**:            3.8.3 for wxPython4, 2.7.17 for wxPython 3, downloaded from Python website

**Description of the problem**:
I see a black bar which moves when dragging to resize a column.  I presume that the bar is intended to preview the new column size, but is being drawn in the wrong place.  For me it is to the left of the cursor.

A similar issue #1031 was raised but closed by the OP with the comment "Must have been an issue in my environment, problem seems to have corrected itself." 

This behaviour can be seen in the wxPython demo.

**Comment by cbeytas** [comment]
Moving displays relative to each other seems to throw off the `wx.ScreenDC()` which is used to draw the column line.
When in a bad state, closing and restarting the wx app won't change anything.
However simply locking the computer and unlocking it puts everything back to normal.
This issue also affects `wx.SplitterWindow` where the `wx.SP_LIVE_UPDATE` flag isn't set.

https://user-images.githubusercontent.com/4748055/105639206-ccc7ed00-5e6e-11eb-990b-ba8547d55de1.mp4

Drawing on a screen DC with the `wx.INVERT` logical function may not be recommended anymore according to the [docs](https://wxpython.org/Phoenix/docs/html/wx.DC.html?highlight=setlogicalfunction#wx.DC.SetLogicalFunction).
The line is only drawn correctly in Win XP/7/10 and GTK2. Under MacOS or GTK3 no line gets drawn.

**Comment by RobinD42** [comment]
Yep, the old hack used for non-live updates really doesn't work well these days. I think there has been some discussion of simply removing it from some of the core widgets in the C++ wxWidgets. I wouldn't mind seeing some PRs to do the same for our generic widgets, or adding live updates for those that may not already have the option.
